client/rest/src/plugins/rosetta/constructionRoutes.js

```diff
@@ -101,5 +123,231 @@ export default {
+
+			aggregateTransaction.cosignatures.forEach(cosignature => {
+				cosignature.signature = new models.Signature(findSignature(cosignature.signerPublicKey));
+			});
+
+			const transactionPayload = aggregateTransaction.serialize();
+
+			const response = new ConstructionCombineResponse();
+			response.signed_transaction = utils.uint8ToHex(transactionPayload);
+			return response;
+		}));
+
+		server.post('/construction/parse', rosettaPostRouteWithNetwork(networkName, ConstructionParseRequest, async typedRequest => {
+			const xymCurrency = new Currency('symbol.xym', 6);
+
+			const modelAddressToString = address => new Address(address.bytes).toString();
+			const publicKeyToAccountIdentifier = publicKey =>
+				new AccountIdentifier(facade.network.publicKeyToAddress(publicKey).toString());
+
+			let id = 0;
+			const operations = [];
+			const allSignerPublicKeyStringSet = new Set();
+			const aggregateTransaction = facade.transactionFactory.static.deserialize(utils.hexToUint8(typedRequest.transaction));
+			aggregateTransaction.transactions.forEach(transaction => {
+				allSignerPublicKeyStringSet.add(transaction.signerPublicKey.toString());
+
+				if (models.TransactionType.TRANSFER.value === transaction.type.value) {
+					const amount = transaction.mosaics[0].amount.value;
+
+					const sendOperation = new Operation(new OperationIdentifier(id++), 'transfer');
+					sendOperation.account = publicKeyToAccountIdentifier(transaction.signerPublicKey);
+					sendOperation.amount = new Amount((-amount).toString(), xymCurrency);
+					operations.push(sendOperation);
+
+					const receiveOperation = new Operation(new OperationIdentifier(id++), 'transfer');
+					receiveOperation.account = new AccountIdentifier(modelAddressToString(transaction.recipientAddress));
+					receiveOperation.amount = new Amount(amount.toString(), xymCurrency);
+					operations.push(receiveOperation);
+				} else if (models.TransactionType.MULTISIG_ACCOUNT_MODIFICATION.value === transaction.type.value) {
+					const operation = new Operation(new OperationIdentifier(id++), 'multisig');
+					operation.account = publicKeyToAccountIdentifier(transaction.signerPublicKey);
+					operation.metadata = {
+						minApprovalDelta: transaction.minApprovalDelta,
+						minRemovalDelta: transaction.minRemovalDelta,
+						addressAdditions: transaction.addressAdditions.map(modelAddressToString),
+						addressDeletions: transaction.addressDeletions.map(modelAddressToString)
+					};
+					operations.push(operation);
+				} else {
+					throw RosettaErrorFactory.NOT_SUPPORTED_ERROR;
+				}
+			});
+
+			aggregateTransaction.cosignatures.forEach(cosignature => {
+				if (allSignerPublicKeyStringSet.has(cosignature.signerPublicKey.toString()))
+					return;
+
+				allSignerPublicKeyStringSet.add(cosignature.signerPublicKey.toString());
+
+				const operation = new Operation(new OperationIdentifier(id++), 'cosign');
```

Comment:
I will need to add cosign to the data side.  currently, I ignore the cosigners. 😅

Reply:
yea, should get for free as part of the operation refactoring